sdk/iot/Azure.Iot.Hub.Service/src/API Design.md

```diff
@@ -70,8 +70,53 @@ APIs for using IotHub v2 jobs
+
 ```csharp
+public class CloudToDeviceMessages
+{
+    /// <summary>
+    /// Send cloud to device message for a device.
+    /// </summary>
+    /// <param name="deviceId">The unique identifier of the device.</param>
+    /// <param name="message">The cloud to device message to be sent to the device. </param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>The Http response.</returns>
+    public virtual async Task<Response> SendMessageAsync(String deviceId, CloudToDeviceMessage message, CancellationToken cancellationToken = default) { }
+
+    /// <summary>
+    /// Retrieve feedback notification for cloud to device messages.
+    /// </summary>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>The message feedback batch, containing the details of the final state of sent messages.</returns>
+    public virtual async Task<Response<MessageFeedbackBatch>> GetMessageFeedbackAsync(CancellationToken cancellationToken = default) { }
+
+    /// <summary>
+    /// Complete a cloud to device feedback message. A completed message is deleted from the service's feedback queue.
+    /// </summary>
+    /// <param name="lockToken">The lock token obtained when the cloud to device message was received, and provided to resolve race conditions when completing a feedback message.
+    /// TODO: lockToken is from the C2D message received on device or from message feedback received on service client?</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>The Http response.</returns>
+    public virtual async Task<Response> CompleteMessageFeedbackAsync(string lockToken, CancellationToken cancellationToken = default) { }
+
+    /// <summary>
+    /// Abandon a cloud to device feedback message. An abandoned message is enqueued back to the service's feedback queue.
+    /// </summary>
+    /// <param name="lockToken">The lock token obtained when the cloud to device message was received, and provided to resolve race conditions when abandoning a feedback message.
+    /// TODO: lockToken is from the C2D message received on device or from message feedback received on service client?</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>The Http response.</returns>
+    public virtual async Task<Response> AbandonMessageFeedbackAsync(string lockToken, CancellationToken cancellationToken = default) { }
+
+    /// <summary>
+    /// Purge the cloud to device message queue for a device.
+    /// </summary>
+    /// <param name="deviceId">The unique identifier of the device.</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>The result of the purge operation.</returns>
+    public virtual async Task<Response<PurgeMessageQueueResult>> PurgeMessageQueueAsync(string deviceId, CancellationToken cancellationToken = default) { }
```

Comment:
suggestions for naming if we are deciding this API to be in same class: 
PurgeC2dMessageQueueAsync
PurgeCloudToDeviceMessageQueueAsync ( I know this is very long but less confusing)